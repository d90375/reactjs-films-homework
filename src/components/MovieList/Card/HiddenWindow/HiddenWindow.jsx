import PropTypes from "prop-types";
import React from "react";
import "./HiddenWindow.scss";

const HiddenWindow = ({ onShowTrailer, onShowInfo }) => {
  return (
    <div className="hiddenWindow">
      <button className="hiddenWindow__playBtn" type="button" onClick={onShowTrailer}>
        â–º
      </button>
      <span className="hiddenWindow__signature">Watch Now</span>
      <button onClick={onShowInfo} className="hiddenWindow__infoBtn" type="button">
        View Info
      </button>
    </div>
  );
};

export default HiddenWindow;

HiddenWindow.propTypes = {
  onShowInfo: PropTypes.func.isRequired,
  onShowTrailer: PropTypes.func.isRequired
};
