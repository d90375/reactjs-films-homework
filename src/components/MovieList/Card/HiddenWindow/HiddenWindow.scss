.hiddenWindow {
  display: none;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  background-color: rgba(0, 0, 0, 0.1);
  border-radius: 5px;
  flex-direction: column;
  align-items: center;
  height: calc(100% - 80px);
  justify-content: center;
  transition: all 0.1s ease-in;
  font-family: "HelveticaNeue", serif;

  &:hover {
    background-color: rgba(0, 0, 0, 0.7);
    display: flex;
  }
}

.hiddenWindow__playBtn {
  background-color: #3bb4e0;
  border: 6px solid #fff;
  border-radius: 50%;
  box-sizing: border-box;
  color: #f5f5f5;
  cursor: pointer;
  display: block;
  font-size: 2rem;
  font-weight: bold;
  height: 80px;
  line-height: 60px;
  margin-bottom: 10px;
  outline: none;
  padding-left: 14px;
  padding-top: 6px;
  text-align: center;
  text-decoration: none;
  transition: all 0.3s ease;
  width: 80px;

  &:hover {
    background-color: #3bb4e0;
    box-shadow: 0 0 10px rgba(255, 255, 255, 1);
    text-shadow: 0 0 10px rgba(255, 255, 255, 1);
  }
}

.hiddenWindow__signature {
  display: block;
  margin-bottom: 25px;
}

.hiddenWindow__infoBtn {
  background: transparent;
  border: 2px solid white;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  height: 30px;
  outline: none;
  width: 100px;

  &:hover {
    background: rgba(159, 159, 159, 0.5);
  }
}
