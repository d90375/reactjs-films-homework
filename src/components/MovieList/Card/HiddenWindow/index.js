import HiddenWindow from "./HiddenWindow";

export default HiddenWindow;
