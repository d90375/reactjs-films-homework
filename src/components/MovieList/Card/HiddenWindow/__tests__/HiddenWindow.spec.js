import React from "react";
import { shallow } from "enzyme";
import HiddenWindow from "../HiddenWindow";

describe("HiddenWindow", () => {
  let component;

  const mockCallBacks = {
    onShowInfo: jest.fn(),
    onShowTrailer: jest.fn()
  };

  it("should render HiddenWindow component", () => {
    component = shallow(<HiddenWindow {...mockCallBacks} />);
    expect(component).toMatchSnapshot();
  });

  describe("should handler", () => {
    it("should call onShowInfo method", () => {
      component = shallow(<HiddenWindow {...mockCallBacks} />);
      expect(mockCallBacks.onShowInfo.mock.calls.length).toBe(0);
      component.find("[data-id='hidden-window-infoBtn']").simulate("click");
      expect(mockCallBacks.onShowInfo.mock.calls.length).toBe(1);
    });

    it("should call onShowTrailer method", () => {
      component = shallow(<HiddenWindow {...mockCallBacks} />);
      expect(mockCallBacks.onShowTrailer.mock.calls.length).toBe(0);
      component.find("[data-id='hidden-window-playBtn']").simulate("click");
      expect(mockCallBacks.onShowTrailer.mock.calls.length).toBe(1);
    });
  });
});
