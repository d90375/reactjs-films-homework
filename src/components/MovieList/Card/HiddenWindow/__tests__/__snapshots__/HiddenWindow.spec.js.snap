// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiddenWindow should render HiddenWindow component 1`] = `
<div
  className="hiddenWindow"
>
  <button
    className="hiddenWindow__playBtn"
    onClick={[MockFunction]}
    type="button"
  >
    â–º
  </button>
  <span
    className="hiddenWindow__signature"
  >
    Watch Now
  </span>
  <button
    className="hiddenWindow__infoBtn"
    onClick={[MockFunction]}
    type="button"
  >
    View Info
  </button>
</div>
`;
